### PR TITLE
Improved QFX Q

### PR DESCRIPTION
**Pull request by JamesVanBoxtel**
Moved over every graphics API to push only if we aren't inside love.draw 
Moved a couple other common cases into draw
Fixed the fps graph to look better and scale properly 
Fixed the order of rendering to make more sense
Fixed the graph to not do order N updates every insertion, instead using a rolling buffer

### CODE REVIEW

**Review by Sankyr — COMMENTED** [review]
why are we branching on GAME.isDrawing for all of these draw calls? are we calling each function twice, once in update and once in draw?

File: RunTimeGraph.lua [hunk 1]
```diff
@@ -59,7 +60,14 @@ function RunTimeGraph:updateWithMetrics(runMetrics)
 end
 
 function RunTimeGraph:draw()
+  love.graphics.push()
+
+  love.graphics.translate(GAME.canvasX, GAME.canvasY)
+  love.graphics.scale(GAME.canvasXScale, GAME.canvasYScale)
+
```
Comment (Endaris) [inline RunTimeGraph.lua:67]: This code is a bit fuzzy to me, is that to fix the displacement issue with the graph?

File: libraries/BarGraph.lua [hunk 2]
```diff
@@ -45,24 +52,29 @@ end
 
 function BarGraph.drawGraphs(graphs)
   local oldFont = love.graphics.getFont()
-  love.graphics.setFont(BarGraph.font)
+  gfx_q:push({love.graphics.setFont, {BarGraph.font}})
```
Comment (Endaris) [inline libraries/BarGraph.lua:55]: Why is the font set in gfx queue when everything else isn't?

File: select_screen/select_screen.lua [hunk 3]
```diff
@@ -973,7 +973,7 @@ function select_screen.main(self, character_select_mode, roomInitializationMessa
 
   -- Main loop for running the select screen and drawing
   while true do
-    graphics:draw(self)
+    gfx_q:push({graphics.draw, {graphics, self}})
```
Comment (Endaris) [inline select_screen/select_screen.lua:976]: clever

File: CustomRun.lua [hunk 4]
```diff
@@ -92,13 +92,6 @@ function CustomRun.innerRun()
     love.graphics.origin()
     love.graphics.clear(love.graphics.getBackgroundColor())
 
-    -- draw the RunTimeGraph here so it doesn't contribute to the love.draw load
-    if CustomRun.runTimeGraph then
-      local preGraphDrawTime = love.timer.getTime()
-      CustomRun.runTimeGraph:draw()
-      CustomRun.runMetrics.graphDuration = CustomRun.runMetrics.graphDuration + (love.timer.getTime() - preGraphDrawTime)
-    end
-
```
Comment (Endaris) [inline CustomRun.lua:101]: I'm genuinely confused why that was here twice.
On my local branch it was there only once until I pulled but then again, there shouldn't have been anything pushed that wasn't local.
I think I'm gonna reset my repo after saving the relevant branches.

File: README.md [hunk 5]
```diff
@@ -46,9 +46,8 @@ Try to follow the following code guidelines when contributing:
 - Make smaller methods
 - Don’t duplicate code, break it into smaller reusable chunks and use that in both spots
 - Writing tests for how the code should work is extremely beneficial
-- Constants should be local to a file / scope unless they need to be shared everywhere
-- Avoid the use of shortlived tables and consider pooling if you can't
```
Comment (Endaris) [inline README.md:50]: why not :-(

Comment (Sankyr) [reply README.md:50]: I agree, these seem like valid guidelines, why remove them?

**Pull request merged**